src/compiler/checker.ts

```diff
@@ -7366,8 +7369,12 @@ namespace ts {
                 if (symbol && symbolIsValue(symbol)) {
                     return symbol;
                 }
-                if (resolved === anyFunctionType || resolved.callSignatures.length || resolved.constructSignatures.length) {
-                    const symbol = getPropertyOfObjectType(globalFunctionType, name);
+                const functionType = resolved === anyFunctionType ? globalFunctionType :

```

Comment:
This should be put into a function and anyFunctionType’s declaration needs a comment saying “Use this function instead”.